crates/telio-wg/src/wg.rs

```diff
@@ -748,7 +867,53 @@ impl State {
 
         self.interface = to.clone();
 
-        success
+        Ok(success)
+    }
+
+    fn compute_new_node_info(
+        &self,
+        old_info: &NoLinkDetectionInfo,
+        new_tx: u64,
+        new_rx: u64,
+    ) -> NoLinkDetectionInfo {
+        let configurable_rtt = match self.no_link_detection {
+            NoLinkDetection::Disabled => DEFAULT_CONFIGURABLE_RTT,
+            NoLinkDetection::Enabled { rtt } => rtt,
+        };
+
+        let now = Instant::now();
+
+        let (rx_bytes, rx_ts) = if new_rx > old_info.last_rx_bytes {
+            (new_rx, now)
+        } else {
+            (old_info.last_rx_bytes, old_info.last_rx_timestamp)
+        };
+
+        let (tx_bytes, tx_ts) = if new_tx > old_info.last_tx_bytes {
+            // Ignore the keepalive messages
```

Comment:
It could be stated that this only works in a limited fashion: because we poll every second, it means the detection might not work as intended if there are more packets within that time window as the difference is the sum of packets.

On the sender side persistent-keepalives might interfere as those are emitted regardless the activity, meaning for example if persistent-keepalive is set at 5seconds then it might interfere with the passive-keepalive which is a constant of 10seconds.

IMHO **This actually could be a testcase: set persistent-keepalive to 5seconds and observe that it reports link as up when receiver is offline. RTT might be 0 to speed it up**. What are your thoughts on the subject? :)